File: mobile/lib/models/player.dart
class Player {
  final int id;
  final int? lobbyId;
  final String name;
  final String? word;

  Player({
    required this.id,
    required this.name,
    this.lobbyId,
    this.word,
  });

  factory Player.fromJson(Map<String, dynamic> json) {
    return Player(
      id: json['id'] as int,
      name: json['name'] as String,
      lobbyId: json['lobby_id'] as int?,
      word: json['word'] as String?,
    );
  }
}

File: backend/app/Models/Player.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Player extends Model
{
    protected $fillable = [
        'lobby_id',
        'name',
        'role',
        'word'
    ];

    public function lobby()
    {
        return $this->belongsTo(Lobby::class);
    }

    public function questionsAsked()
    {
        return $this->hasMany(Question::class, 'asker_id');
    }

    public function questionsReceived()
    {
        return $this->hasMany(Question::class, 'target_id');
    }
}
